x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/task/retrievers/__mocks__/mocks.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RuleMigrationsRetriever } from '..';
+import type { RuleMigrationsDataClient } from '../../../data/rule_migrations_data_client';
+
+export const createRuleMigrationsRetrieverMock = () => {
+  const mockResources = {
+    initialize: jest.fn().mockResolvedValue(undefined),
+    getResources: jest.fn().mockResolvedValue({}),
+    migrationId: 'test-migration-id',
+    dataClient: {} as RuleMigrationsDataClient,
```

Comment:
Do we need these?

Reply:
These was required to satisfy the type, as they are not optional properties of the original class mostly, I can remove it if there is a better way.

Reply:
They are private attributes, we should not need to define them. Maybe using the `PublicMethodsOf` type util will help